projects/packages/password-checker/src/class-password-checker.php

```diff
@@ -417,63 +407,77 @@ protected function test_not_same_as_other_user_data( $password, $strings_to_test
 				return false;
 			}
 		}
+
 		return true;
 	}
 
 	/**
 	 * A shorthand for the not in array construct.
 	 *
-	 * @param Mixed $needle the needle.
+	 * @param mixed $needle   the needle.
 	 * @param array $haystack the haystack.
-	 * @return is the needle not in the haystack?
+	 *
+	 * @return bool is the needle not in the haystack?
 	 */
 	protected function negative_in_array( $needle, $haystack ) {
-		if ( in_array( $needle, $haystack, true ) ) {
-			return false;
-		}
-
-		return true;
+		return ! in_array( $needle, $haystack, true );
 	}
 
 	/**
 	 * A helper function used to break a single string into its constituents so
 	 * that both the full string and its constituents and any variants thereof
 	 * can be tested against the password.
 	 *
-	 * @param String $string the string to be broken down.
-	 * @param String $explode_delimiter delimiter.
-	 * @return NULL|Array array of fragments, or NULL on empty string.
+	 * @param string $string            the string to be broken down.
+	 * @param string $explode_delimiter delimiter.
+	 *
+	 * @return bool
 	 */
 	protected function add_user_strings_to_test( $string, $explode_delimiter = ' ' ) {
-
 		// Don't check against empty strings.
 		if ( empty( $string ) ) {
-			return;
+			return false;
 		}
 
 		$strings = explode( $explode_delimiter, $string );
 
 		// Remove any non alpha numeric characters from the strings to check against.
 		foreach ( $strings as $key => $_string ) {
-			$strings[ $key ] = preg_replace( '/[^a-zA-Z0-9]/', '', $_string );
+			$strings[ $key ] = trim( preg_replace( '/[^a-zA-Z0-9]/', '', $_string ) );
```

Comment:
This might wind up adding an empty string to the array.

Reply:
Nice catch!! Fixed.

Reply:
Huh, I thought I deleted that comment. It doesn't matter if an empty string gets added here because the `array_filter()` below will remove it.